app/controllers/cases_controller.rb

```diff
@@ -81,6 +82,25 @@ def all_activity
     @activities = CaseActivity.all
   end
 
+  def rename_file
+
+    @case = Case.find(params[:id])
+    file = @case.files.find{params["file_id"]}
+    new_filename = params["new_filename"]
+
+    filenames = @case.files.map{|f| f.filename}
+
+    if not new_filename == file.filename and filenames.include?(new_filename)
```

Comment:
I think it's not necessary to check the file.filename as it's included in `filenames`